src/os_crypto/sha1/sha1_op.c

```diff
@@ -192,35 +198,39 @@ int OS_SHA1_File_Nbytes_with_fp_check(const char * fname, SHA_CTX * c, os_sha1 o
         }
 
         buf[n] = '\0';
-        SHA1_Update(c, buf, n);
+        EVP_DigestUpdate(*c, buf, n);
     }
 
-    SHA_CTX aux = *c;
+    EVP_MD_CTX *aux = EVP_MD_CTX_new();
+    EVP_MD_CTX_copy(aux, *c);
 
-    SHA1_Final(&(md[0]), &aux);
+    EVP_DigestFinal(aux, md, NULL);
 
     OS_SHA1_Hexdigest(md, output);
 
+    EVP_MD_CTX_free(aux);
+
     fclose(fp);
 
     return (0);
 }
 
-void OS_SHA1_Stream(SHA_CTX *c, os_sha1 output, char * buf) {
+void OS_SHA1_Stream(EVP_MD_CTX *c, os_sha1 output, char * buf) {
     if(buf) {
         size_t n = strlen(buf);
 
-        SHA1_Update(c, buf, n);
+        EVP_DigestUpdate(c, buf, n);
```

Comment:
Same, here also the context is already created and initialized. Explain this.

Reply:
Comment added to the function header